pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -87,6 +87,11 @@ func init() {
 	cloudprovider.RegisterCloudProvider(ProviderName, func(config io.Reader) (cloudprovider.Interface, error) { return newGCECloud(config) })
 }
 
+// Raw access to the underlying GCE service, probably should only be used for e2e tests
```

Comment:
I like raw access to resources so downstream controllers are free to define their own interfaces. ListForwardingRules is only used by L4 loadbalancers though, which already has a TCPLoadBalancer interface that's more tightly coupled with the servicecontroller. List*Global*ForwardingRules is used for L7 and there's no coupling. I think a big difference between these 2 is that configuring an L4 is kind of similar on AWS and GCE (if you squint), so an interface makes more sense. They're very different at L7.